File: scala/src/main/scala/com/authdog/types/Email.scala
package com.authdog.types

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}

/**
 * User email
 */
case class Email(
  id: String,
  value: String,
  `type`: Option[String]
)

object Email {
  implicit val decoder: Decoder[Email] = deriveDecoder[Email]
  implicit val encoder: Encoder[Email] = deriveEncoder[Email]
}

File: planned/php/src/Types/Email.php
<?php

namespace Authdog\Types;

/**
 * User email
 */
class Email
{
    public string $id;
    public string $value;
    public ?string $type;

    public function __construct(array $data)
    {
        $this->id = $data['id'] ?? '';
        $this->value = $data['value'] ?? '';
        $this->type = $data['type'] ?? null;
    }

    public function getAddress(): string
    {
        return $this->value;
    }
}
